// ems-chain-js/dist/codegen/ems/v1/state.mjs
import { BinaryReader, BinaryWriter } from "../../binary";
import { bytesFromBase64, base64FromBytes } from "../../helpers";
function createBaseExampleData() {
    return {
        account: new Uint8Array(),
        amount: BigInt(0)
    };
}
export const ExampleData = {
    typeUrl: "/ems.v1.ExampleData",
    encode(message, writer = BinaryWriter.create()) {
        if (message.account.length !== 0) {
            writer.uint32(10).bytes(message.account);
        }
        if (message.amount !== BigInt(0)) {
            writer.uint32(16).uint64(message.amount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseExampleData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.account = reader.bytes();
                    break;
                case 2:
                    message.amount = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseExampleData();
        message.account = object.account ?? new Uint8Array();
        message.amount = object.amount !== undefined && object.amount !== null ? BigInt(object.amount.toString()) : BigInt(0);
        return message;
    },
    fromAmino(object) {
        const message = createBaseExampleData();
        if (object.account !== undefined && object.account !== null) {
            message.account = bytesFromBase64(object.account);
        }
        if (object.amount !== undefined && object.amount !== null) {
            message.amount = BigInt(object.amount);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.account = message.account ? base64FromBytes(message.account) : undefined;
        obj.amount = message.amount !== BigInt(0) ? (message.amount?.toString)() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return ExampleData.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return ExampleData.decode(message.value);
    },
    toProto(message) {
        return ExampleData.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/ems.v1.ExampleData",
            value: ExampleData.encode(message).finish()
        };
    }
};
