// ems-chain-js/dist/codegen/ems/v1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/ems.v1.MsgUpdateParams", tx_1.MsgUpdateParams], ["/ems.v1.MsgCreateEventParams", tx_1.MsgCreateEventParams], ["/ems.v1.MsgIssueEventNFTParams", tx_1.MsgIssueEventNFTParams], ["/ems.v1.MsgAddEventOrganizerParams", tx_1.MsgAddEventOrganizerParams], ["/ems.v1.MsgUpdateEventStatusParams", tx_1.MsgUpdateEventStatusParams]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        updateParams(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        },
        msgCreateEvent(value) {
            return {
                typeUrl: "/ems.v1.MsgCreateEventParams",
                value: tx_1.MsgCreateEventParams.encode(value).finish()
            };
        },
        msgIssueEventNFT(value) {
            return {
                typeUrl: "/ems.v1.MsgIssueEventNFTParams",
                value: tx_1.MsgIssueEventNFTParams.encode(value).finish()
            };
        },
        msgAddEventOrganizer(value) {
            return {
                typeUrl: "/ems.v1.MsgAddEventOrganizerParams",
                value: tx_1.MsgAddEventOrganizerParams.encode(value).finish()
            };
        },
        msgUpdateEventStatus(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateEventStatusParams",
                value: tx_1.MsgUpdateEventStatusParams.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        updateParams(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateParams",
                value
            };
        },
        msgCreateEvent(value) {
            return {
                typeUrl: "/ems.v1.MsgCreateEventParams",
                value
            };
        },
        msgIssueEventNFT(value) {
            return {
                typeUrl: "/ems.v1.MsgIssueEventNFTParams",
                value
            };
        },
        msgAddEventOrganizer(value) {
            return {
                typeUrl: "/ems.v1.MsgAddEventOrganizerParams",
                value
            };
        },
        msgUpdateEventStatus(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateEventStatusParams",
                value
            };
        }
    },
    fromPartial: {
        updateParams(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        },
        msgCreateEvent(value) {
            return {
                typeUrl: "/ems.v1.MsgCreateEventParams",
                value: tx_1.MsgCreateEventParams.fromPartial(value)
            };
        },
        msgIssueEventNFT(value) {
            return {
                typeUrl: "/ems.v1.MsgIssueEventNFTParams",
                value: tx_1.MsgIssueEventNFTParams.fromPartial(value)
            };
        },
        msgAddEventOrganizer(value) {
            return {
                typeUrl: "/ems.v1.MsgAddEventOrganizerParams",
                value: tx_1.MsgAddEventOrganizerParams.fromPartial(value)
            };
        },
        msgUpdateEventStatus(value) {
            return {
                typeUrl: "/ems.v1.MsgUpdateEventStatusParams",
                value: tx_1.MsgUpdateEventStatusParams.fromPartial(value)
            };
        }
    }
};
